tensornetwork.py

```diff
@@ -521,60 +523,59 @@ def _remove_trace_edge(self, edge: Edge, new_node: Node) -> None:
       new_node.add_edge(e, i)
     self.nodes_set.remove(edge.node1)
 
-  def _remove_edge(self, edge: Edge, new_node: Node) -> None:
-    """Collapse an edge in the network.
+  def _remove_edges(self, edges, node1: Node, node2: Node, new_node: Node) -> None:
+    """Collapse a list of edges shared by two nodes in the network.
 
-    Collapses an edge and updates the rest of the network.
+    Collapses the edges and updates the rest of the network.
 
     Args:
-      edge: The edge to contract.
+      edges: The edges to contract.
       new_node: The new node that represents the contraction of the two old
         nodes.
 
     Raises:
       Value Error: If edge isn't in the network.
     """
-    # Assert that the edge isn't a dangling edge.
-    if edge.is_dangling():
-      raise ValueError("Attempted to remove dangling edge '{}'.".format(edge))
-    if edge.node1 is edge.node2:
-      self._remove_trace_edge(edge, new_node)
-    # Collapse the nodes into a new node and remove the edge.
-    node1 = edge.node1
-    node2 = edge.node2
-    node1_edges = edge.node1.edges[:]
-    node2_edges = edge.node2.edges[:]
-    node1_axis = edge.axis1
-    node2_axis = edge.axis2
-    # Redefine all other edges.
-    num_added_front_edges = len(node1_edges) - 1
-    for i, tmp_edge in enumerate(node1_edges[:node1_axis]):
-      tmp_edge.update_axis(
-          old_axis=i, old_node=node1, new_axis=i, new_node=new_node)
-    for i, tmp_edge in enumerate(node1_edges[node1_axis + 1:]):
-      tmp_edge.update_axis(
-          old_axis=i + node1_axis + 1,
+    if node1 is node2:
+      raise ValueError(
+        "Trace edge '{}' cannot be removed by  _remove_edges().".format(
+          edges[0]))
+
+    node1_edges = node1.edges[:]
+    node2_edges = node2.edges[:]
+
+    nodes_set = set([node1, node2])
+    for edge in edges:
+      if edge.is_dangling():
+        raise ValueError("Attempted to remove dangling edge '{}'.".format(edge))
+      if set([edge.node1, edge.node2]) != nodes_set:
+        raise ValueError(
+          "Attempted to remove edges belonging to different node pairs: "
+          "'{}' != '{}'.".format(nodes_set, set([edge.node1, edge.node2])))
+  
+    remaining_edges = []
+    for (i, edge) in enumerate(node1_edges):
+      if edge not in edges:  # FIXME: Makes the cost quadratic in # edges
```

Comment:
cast edges to a set() before this loop and use that.

Reply:
Done.